plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadataFactory.java

```diff
@@ -45,6 +46,7 @@ public IcebergMetadataFactory(
 
     public IcebergMetadata create(ConnectorIdentity identity)
     {
-        return new IcebergMetadata(typeManager, commitTaskCodec, catalogFactory.create(identity), hdfsEnvironment);
+        TrinoCatalog trinoCatalog = catalogFactory.create(identity);
+        return new IcebergMetadata(typeManager, commitTaskCodec, trinoCatalog, hdfsEnvironment);
```

Comment:
revert or separate commit

Reply:
leftover from a previous attempt to provide a separate redirect table functionality.